frontend/packages/operator-lifecycle-manager/src/components/operand/index.tsx

```diff
@@ -64,10 +72,99 @@ import { CustomResourceDefinitionModel } from '@console/internal/models';
 import { useK8sModels } from '@console/shared/src/hooks/useK8sModels';
 import { DescriptorDetailsItem, DescriptorDetailsItemList } from '../descriptors';
-import { useTranslation } from 'react-i18next';
 import { isMainStatusDescriptor } from '../descriptors/utils';
 import { DescriptorConditions } from '../descriptors/status/conditions';
 
+const EmptyMsg = () => (
+  <MsgBox
+    title={i18next.t('olm~No provided APIs defined')}
+    detail={i18next.t('olm~This application was not properly installed or configured.')}
+  />
+);
+
+type WrappedToggleViewProps = {
+  toggleValue: ToggleValue;
+  toggleView?: boolean;
+  title?: string;
+  onChangeToggleValue: (newValue: ToggleValue) => void;
+};
+const WrappedToggleView: React.FC<WrappedToggleViewProps> = ({
+  toggleValue,
+  title,
+  onChangeToggleValue,
+}) => {
+  const { t } = useTranslation();
+  return (
+    <div className="co-operator-details__toggle-view">
+      <span className="co-operator-details__toggle-view-heading">
+        <SectionHeading text={title || t('olm~All Instances')} />
+      </span>
+      <ToggleView
+        value={toggleValue}
+        onChange={onChangeToggleValue}
+        label={t('olm~Show operands in:')}
+        optionTitles={[t('olm~All namespaces'), t('olm~Current namespace only')]}
+      />
+    </div>
+  );
+};
+
+const firehoseResources = (models, providedAPIs: ProvidedAPI[], namespaced: boolean) => {
+  // Exclude provided APIs that do not have a model
+  return providedAPIs.reduce((resourceAccumulator, api) => {
+    const reference = referenceForProvidedAPI(api);
+    const model = models?.[reference];
+    return model
+      ? [
+          ...resourceAccumulator,
+          namespaced
+            ? {
+                kind: referenceForProvidedAPI(api),
+                namespaced: model.namespaced,
+                prop: api.kind,
+              }
+            : {
+                kind: referenceForProvidedAPI(api),
+                prop: api.kind,
+              },
+        ]
+      : resourceAccumulator;
+  }, []);
+};
+
+const flattenResources = (providedAPIs: K8sResourceKind[]) => {
+  const owners = (ownerRefs: OwnerReference[], items: K8sResourceKind[]) =>
+    ownerRefs.filter(({ uid }) => items.filter(({ metadata }) => metadata.uid === uid).length > 0);
+
+  return (resources: { [kind: string]: { data: K8sResourceKind[] } }) =>
+    _.flatMap(resources, (resource) => _.map(resource.data, (item) => item)).filter(
+      ({ kind, metadata }, i, allResources) =>
+        providedAPIs.filter((item) => item.kind === kind).length > 0 ||
+        owners(metadata.ownerReferences || [], allResources).length > 0,
+    );
+};
+
+const hasOtherManagedNamespaces = (obj: K8sResourceKind) => {
+  const olmTargetNamespaces = obj?.metadata?.annotations?.['olm.targetNamespaces'] ?? '';
+  const managedNamespaces = olmTargetNamespaces?.split(',') || [];
+  return managedNamespaces.length === 1 && managedNamespaces[0] === '';
```

Comment:
This return logic seems to indicate "does not have OtherManagedNamespaces", or "has one ns === ' ' " ?